shared/dataflow/codeql/dataflow/internal/DataFlowImpl.qll

```diff
@@ -3985,11 +3994,15 @@ module MakeImpl<InputSig Lang> {
       PathNodeMid mid, NodeEx node, FlowState state, CallContext cc, SummaryCtx sc, DataFlowType t,
       AccessPath ap
     ) {
-      exists(DataFlowType t0 |
-        pathStep0(mid, pragma[only_bind_into](node), pragma[only_bind_into](state), cc, sc, t0, ap) and
+      exists(DataFlowType t0, boolean isStoreStep |
+        pathStep0(mid, pragma[only_bind_into](node), pragma[only_bind_into](state), cc, sc, t0, ap,
+          isStoreStep) and
         Stage5::revFlow(pragma[only_bind_into](node), pragma[only_bind_into](state), ap.getApprox()) and
         strengthenType(node, t0, t) and
         not inBarrier(node, state)
+      |
+        isStoreStep = true or
+        not ap.storeTargetIsClearedAt(node)
```

Comment:
Have you checked the resulting RA for this? It looks like something that'll lead to pipeline duplication, i.e. tuples that satisfy both branches will get duplicated and cause additional de-duplication work at the end. Rewording this as an `if-then-else` is likely better.
```suggestion
        if isStoreStep = true then any() else not ap.storeTargetIsClearedAt(node)
```

Reply:
Or (since `isStoreStep` is mostly false anyway) the following might be nicer:
```suggestion
        if ap.storeTargetIsClearedAt(node) then isStoreStep = true else any()
```